version/current_test.go

```diff
@@ -31,13 +32,23 @@ func (*CurrentSuite) TestCurrentSeries(c *gc.C) {
 		case "windows":
 			c.Check(s, gc.Matches, `win2012hvr2|win2012hv|win2012|win2012r2|win8|win81|win7`)
 		default:
-			c.Assert(s, gc.Equals, "n/a")
+			if s != "n/a" {
+				current_os, err := version.GetOSFromSeries(s)
+				c.Assert(err, gc.IsNil)
+				// There is no lsb_release command on CentOS.
+				switch current_os {
+				case version.CentOS:
+					c.Check(s, gc.Matches, `centos7`)
+				default:
+					c.Assert(string(out), gc.Equals, "Codename:\t"+s+"\n")
+				}
+			}
 		}
 	} else {
-		os, err := version.GetOSFromSeries(s)
+		current_os, err := version.GetOSFromSeries(s)
 		c.Assert(err, gc.IsNil)
 		// There is no lsb_release command on CentOS.
-		switch os {
+		switch current_os {
```

Comment:
this test doesn't make sense to me. we search for centos in both branches. lsb_release will either error out or it won't, it can't be both.

I would pull the call to getosfromseries outside of the if, and just switch on version.Windows, version.CentOS, etc instead of runtime.GOOS. You need to test this on all 3 platforms though.